core/src/main/java/org/apache/struts2/interceptor/csp/CspSettings.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.struts2.interceptor.csp;
+
+import javax.servlet.http.HttpServletResponse;
+
+public interface CspSettings {
+
+    String CSP_ENFORCE_HEADER = "Content-Security-Policy";
+    String CSP_REPORT_HEADER = "Content-Security-Policy-Report-Only";
+    String OBJECT_SRC = "object-src";
+    String SCRIPT_SRC = "script-src";
+    String BASE_URI = "base-uri";
+    String REPORT_URI = "report-uri";
+    String NONE = "none";
+    String STRICT_DYNAMIC = "strict-dynamic";
+    String HTTP = "http:";
+    String HTTPS = "https:";
+
+    void addCspHeaders(HttpServletResponse response);
+    void setReportUri(String uri);
+    void setReporting(boolean value);
```

Comment:
it might not be entirely clear what the relationship between ReportUri and Reporting is. Let's make this more explicit by renaming `setReporting` (and associated fields) to `setReportMode` please